edb/server/protocol/auth_ext/base.py

```diff
@@ -86,23 +100,36 @@ async def exchange_code(
         async with self.http_factory(
             base_url=f"{token_endpoint.scheme}://{token_endpoint.netloc}"
         ) as client:
-            resp = await client.post(
-                token_endpoint.path,
-                json={
-                    "grant_type": "authorization_code",
-                    "code": code,
-                    "client_id": self.client_id,
-                    "client_secret": self.client_secret,
-                    "redirect_uri": redirect_uri,
-                },
-            )
+            request_body = {
+                "grant_type": "authorization_code",
+                "code": code,
+                "client_id": self.client_id,
+                "client_secret": self.client_secret,
+                "redirect_uri": redirect_uri,
+            }
+            if self.content_type == ContentType.JSON:
+                resp = await client.post(
+                    token_endpoint.path,
+                    json=request_body,
+                    headers={"Accept": "application/json"},
```

Comment:
This can be a common headers dict shared by the 2 if/else branches I think?

```suggestion
                    headers={"Accept": self.content_type.value},
```